lib/tests/test_stdlib.metta

```diff
@@ -49,3 +49,43 @@
 !(assertEqualToResult
    (id (noeval (+ 1 2)))
    (3))
+
+; assert
+
+!(assert True)
+!(assert (or True False))
+!(assertEqual (assert (and True False)) (Error (assert (and True False)) ((and True False) not True)))
+
+; asserts with custom messages
+; first part of tests is to check if asserts works as intended when assert should output [()].
+; So those tests are duplicate of tests for original asserts
+
+!(assertEqualToResultMsg
+   (noeval (+ 1 2))
+   ((+ 1 2))
+   "Test message")
+
+!(assertEqualToResultMsg
+   (= (f) (+ 1 2))
+   ((= (f) 3))
+   "Test message")
+
+(= (f1 $x) $x)
+(= (f2 $x) $x)
+
+!(assertEqualMsg (+ 1 2) 3 "Test message")
+!(assertAlphaEqualMsg (f1 $x) (f2 $x) "Test message")
+
+(= (f3) $x)
+!(assertAlphaEqualToResultMsg ((f3) (f3)) (($x $y)) "Test message")
+
+; next part of tests is to test if message is being output on assert failure
+
+!(assertEqual (assertEqualMsg (+ 1 2) 4 "Test message") ("Test message"))
```

Comment:
Do I understand correctly that `(assertEqualMsg (+ 1 2) 4 "Test message")` returns `("Test message")`? I supposed it should return a `Error` expression with the given message. Otherwise, it doesn't work as an assert?

Reply:
@Necr0x0Der well currently yes. As i said in the description:
```
One thing about asserts with custom message. Regular assert will stop program from further interpretation on failure, but -Msg asserts won't. I'm not sure about this behavior. As an alternative, instead of just outputting custom message it could be wrapped into Error like regular output of assert. So some
!(assertEqualMsg (+ 1 2) 4 "Test message") will output (Error (assertEqualMsg (+ 1 2) 4 "Test message") "Test message") instead of just ("Test message"). What behavior is preferable?
```